packages/hash/frontend/src/components/block-loader/block-loader.tsx

```diff
@@ -50,53 +47,105 @@ type BlockLoaderProps = {
  */
 export const BlockLoader: FunctionComponent<BlockLoaderProps> = ({
   blockEntityId,
+  blockEntityTypeId,
   blockMetadata,
-  blockSchema,
   editableRef,
-  entityId,
-  entityTypeId,
-  entityProperties,
   onBlockLoaded,
   // shouldSandbox,
+  wrappingEntityId,
 }) => {
   const isReadonlyMode = useIsReadonlyMode();
   const { aggregateEntities } = useBlockProtocolAggregateEntities();
+  const { updateEntity } = useBlockProtocolUpdateEntity();
   const { uploadFile } = useBlockProtocolFileUpload(isReadonlyMode);
-
-  const { showDataMappingUi, setShowDataMappingUi } = useBlockContext();
-
-  // Storing these in local storage is a temporary solution – we want them in the db soon
-  // Known issue: this hook always sets _some_ value in local storage, so we end up with unnecessary things stored there
-  const mapId = `${entityTypeId}:${blockMetadata.source}`;
-  const [schemaMap, setSchemaMap] = useLocalstorageState<SchemaMap>(
-    `map:${mapId}`,
-    { mapId },
-  );
-
-  const graphProperties = useMemo<
-    Required<BlockGraphProperties<UnknownRecord>["graph"]>
-  >(() => {
-    const convertedEntityTypesForProvidedEntities: BpEntityType[] = [];
-
-    const blockEntity = convertApiEntityToBpEntity({
-      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- improve logic or types to remove this comment
-      entityId: entityId ?? "entityId-not-yet-set", // @todo ensure blocks always get sent an entityId
-      entityTypeId,
-      properties: entityProperties,
-    });
-
-    return {
-      blockEntity,
-      entityTypes: uniqBy(
-        convertedEntityTypesForProvidedEntities,
-        "entityTypeId",
-      ),
-      readonly: isReadonlyMode,
-      // We currently don't construct a block graph or linked aggregations.
-      blockGraph: {} as BlockGraph,
-      linkedAggregations: [],
+  const [graphProperties, setGraphProperties] = useState<Required<
+    BlockGraphProperties["graph"]
+  > | null>(null);
+
+  const { getEntity } = useBlockProtocolGetEntity();
+
+  useEffect(() => {
+    const depths: GraphResolveDepths = {
+      hasRightEntity: {
+        incoming: 2,
+        outgoing: 2,
+      },
+      hasLeftEntity: {
+        incoming: 2,
+        outgoing: 2,
+      },
     };
-  }, [entityId, entityProperties, entityTypeId, isReadonlyMode]);
+
+    if (!blockEntityId) {
+      // when inserting a block in the frontend we don't yet have an entity associated with it
+      // there's a delay while the request to the API to insert it is processed
+      // @todo some better way of handling this – probably affected by revamped collab.
+      //    or could simply not load a new block until the entity is created?
+      const now: string = new Date().toISOString();
+      const placeholderEntity = {
+        metadata: {
+          editionId: {
+            baseId: "placeholder-account%entity-id-not-set",
```

Comment:
I see the todo above and I really think we should consider not loading it until an entity is created. (Not asking for us to change it in this PR as I don't have a good sense at how much effort would be involved). Just this code feels exactly like the kind of code that will cause really annoying bugs, we're setting random placeholder strings (which are not UUIDs and may cause validation failures, although we wouldn't want them to masquerade as being valid to be fair), inside a pretty constrained template-string type, which will give us false confidence.

Reply:
I agree we need to fix this. Something we've spoken about occasionally is the FE being able to generate its own ids, so it can start to refer to entities without having a response from the db. We've planned to consider this in the context of new collab – [task](https://app.asana.com/0/1201095311341924/1203257689029607/f) (internal). I think this would be better than a loading state, although we could implement the latter if the former proved impossible